lib/round.rb

```diff
@@ -0,0 +1,46 @@
+class Round 
+  attr_accessor :deck, 
+                :turns, 
+                :current_card, 
+                :number_correct, 
+                :take_turn,    
+                :turn_count          
+  def initialize(deck)
+    @deck = deck
+    @turns = []
+    @number_correct = 0
+    @turn_count = 0
+  end 
+  def current_card
+    self.deck.cards[self.turns.count]
+  end 
+  def take_turn(guess)
+    new_turn = Turn.new(guess, self.current_card)
+    @turn_count += 1
+    @turns << new_turn
+    if new_turn.correct?
+      @number_correct += 1 
+    end 
+    new_turn
+  end
+  def number_correct
+    @number_correct
+  end 
```

Comment:
This method is doing exactly the same thing as our `attr_reader` for `number_correct` so we actually can get rid of this method altogether.